.gitignore

```diff
@@ -89,3 +89,13 @@ Builds/VisualStudio2015/*.sdf
 
 # MSVC
 *.pdb
+
+# cmake output - ignored for now
```

Comment:
Adding these ignores implies the user is doing an in-source build. We don't want to encourage that. In fact, I'd be in favor of disabling in source builds:
```
if("${PROJECT_SOURCE_DIR}" STREQUAL "${PROJECT_BINARY_DIR}")
   message(FATAL_ERROR "In-source builds are not allowed.")
endif()
```
I'd like to get rid of these ignores and maybe add documentation encouraging running cmake in already ignored `build` direcotyr.